fastparquet/api.py

```diff
@@ -416,19 +416,23 @@ def to_pandas(self, columns=None, categories=None, filters=[],
             inferred from the metadata (if this was originally pandas data); if
             the metadata does not exist or index is False, index is simple
             sequential integers.
-        row_filter: bool
+        row_filter: bool or tuple of a list and a ndarray
             Whether filters are applied to whole row-groups (False, default)
             or row-wise (True, experimental). The latter requires two passes of
             any row group that may contain valid rows, but can be much more
             memory-efficient, especially if the filter columns are not required
             in the output.
+            If a tuple, the first element has to be the list of row groups to
+            retrieve, and the second element has to be a ndarray of booleans
+            specifying row per row if it is has to be loaded or not. This
+            data will only be taken into account if 'filters' parameter remains
```

Comment:
Maybe `if filters and not row_filter in [True, False]` we raise a `TypeError`?

Reply:
Hi @martindurant 
I took this feedback into account in the last commit, but I think we can do without raising an exception.
So instead, the checking now makes sure that `if filters and row_filter is True`, then the classic path is taken, else the custom row filtering is retained.
In this latter casr, if `row_filter` is not of the expected type (has no `len` attribute), python will naturally raise a `TypeError`.

I updated the docstring accordingly. In the 2nd case, `filters` is now simply ignored, whatever it is.